packages/strapi-admin/ee/admin/containers/Roles/CreatePage/ContentTypeCollapse/CollapsePropertyMatrix/index.js

```diff
@@ -0,0 +1,49 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { Padded } from '@buffetjs/core';
+import generateHeadersFromActions from './utils/generateHeadersFromActions';
+import Header from './Header';
+import ActionRow from './ActionRow';
+import Wrapper from './Wrapper';
+
+const CollapsePropertyMatrix = ({
+  availableActions,
+  isLast,
+  isOdd,
+  label,
+  propertyName,
+  values,
+}) => {
+  const propertyActions = useMemo(
```

Comment:
Not sure, but I think it's not necessary neither, `ActionRow` and `Header ` will re-render anytime this component will re-render anyways.

Is the `generateHeadersFromActions ` heavy in computations? I mean, the size of the input and the size of the output?